src/renovation_tracker/main.py

```diff
@@ -1,38 +1,115 @@
-from __future__ import annotations
+from fastapi import FastAPI, Depends
+from database import engine, Session
+from enum import StrEnum
+import models
+from pydantic import BaseModel, HttpUrl, Field
+from pydantic_models import *
+from typing import Annotated
+import httpx
+
+
+
+models.Base.metadata.create_all(bind = engine)
+def get_db():
+    db = Session()
+    try:
+        yield db
+    finally:
+        db.close()
+
+db_dependency = Annotated[Session, Depends(get_db)]
+api = FastAPI()
+
+# CREATE Listings
+@api.post("/listings/", response_model=ListingRead)
+async def create_listing(listing:Listing, db: db_dependency):
```

Comment:
You shouldn't have a variable as a function parameter like this; instead include the type declaration explicitly: 
```suggestion
async def create_listing(listing:Listing, db: Annotated[Session, Depends(get_db)]):
```